internal/versions/versions.go

```diff
@@ -115,21 +118,34 @@ func (h *handler) GetReleaseImage(ctx context.Context, openshiftVersion, cpuArch
 	if err := h.kubeClient.List(ctx, clusterImageSets); err != nil {
 		return nil, err
 	}
+	var wg sync.WaitGroup
 	for _, clusterImageSet := range clusterImageSets.Items {
-		existsInCache := false
-		for _, releaseImage := range h.releaseImages {
-			if releaseImage.URL != nil && *releaseImage.URL == clusterImageSet.Spec.ReleaseImage {
-				existsInCache = true
-				break
-			}
+		if err := h.sem.Acquire(ctx, 1); err != nil {
```

Comment:
Maybe we can use something like [samber/lo/parallel](https://github.com/samber/lo/tree/master/parallel) to handle this for us. Haven't tried it, but seems it supports a parallel processing of ForEach: https://github.com/samber/lo/#foreach (looks like just a simple use of WaitGroup though https://github.com/samber/lo/blob/master/parallel/slice.go#L30)

Reply:
We have two separate concerns here.

The wait group is so that we can track when all the routines are finished.
The semaphore is to limit the number of routines that can run concurrently.

I think we need both and that library only handles one.